bottlecap/src/bin/bottlecap/main.rs

```diff
@@ -429,16 +543,46 @@ async fn extension_loop_active(
             shutdown = handle_next_invocation(next_response, invocation_processor.clone()).await;
         } else {
             //Periodic flush scenario, flush at top of invocation
-            if flush_control.should_periodic_flush() {
-                // Should flush at the top of the invocation, which is now
-                flush_all(
+            if current_flush_decision == FlushDecision::Continuous && !last_continuous_flush_error {
+                let tf = trace_flusher.clone();
+                shutdown_handles
+                    .await_flush_handles(&logs_flusher.clone(), &tf, &metrics_flusher)
+                    .await;
+
+                let val = logs_flusher.clone();
+                shutdown_handles
+                    .log_flush_handles
+                    .push_back(tokio::spawn(async move { val.flush(None).await }));
+                let traces_val = trace_flusher.clone();
+                shutdown_handles
+                    .trace_flush_handles
+                    .push_back(tokio::spawn(async move {
+                        traces_val.flush(None).await.unwrap_or_default()
+                    }));
+                let cloned_metrics_flusher = metrics_flusher.clone();
+                shutdown_handles
+                    .metric_flush_handles
+                    .push_back(tokio::spawn(async move {
+                        cloned_metrics_flusher
+                            .lock()
+                            .await
+                            .flush()
+                            .await
+                            .unwrap_or_default()
+                    }));
+                race_flush_interval.reset();
+            } else if current_flush_decision == FlushDecision::Periodic {
```

Comment:
Would it make sense to have a function in the shutdown handles which internally spawns the task and clones the value?